app/grandchallenge/hanging_protocols/models.py

```diff
@@ -88,6 +89,13 @@ class ImagePort(models.TextChoices):
             "label": {
                 "type": "string",
             },
+            "orientation": {
+                "type": "string",
+                "enum": [
+                    orientation.lower()
+                    for orientation in WorkstationConfig.Orientation.labels
```

Comment:
Why? It makes sense to me, I think.

Reply:
It seems like plucking from something deep in other code because it just happens to be the same at the time of writing. Indicative of the required `.lower()` already.

Reply:
I think that it's fine as you're setting the same thing in both cases. The `lower()` is indeed a problem with the design - the labels are the human readable things and so we should be free to change them (think about translations if they're ever implemented). The values are the thing that is stored in the db and should be used for serialization as those things do not change. Same goes for the ImagePort below.

Reply:
CIRRUS does a lower on the `default_orientation` of the `WorkstationConfig`. Serialization of the `WorkstationConfig.default_orientation` is currently done using `get_default_image_interpolation_display` as src. Which I reckon returns the human-readable label. 

There are a whole bunch of these constructs, using labels for serialization in the WorkstationConfig. Either:

1. we ignore it: translations are unlikely to ever arrive
2. we update CIRRUS to work with both DB values and labels, then switch to DB values on GC serialization, then remove the label construct from CIRRUS.
3. we add a mapping in GC to go from DB values to serialized values (e.g. 'A' -> 'axial')

Reply:
Yes, the way to do it is to define your own TextChoice, something like:

```python
class OrientationChoices(TextChoice):
  axial = "axial", "Axial"
  ...

"enum": OrientationChoices.values
```

Reply:
That would replace the problem of using human-readable labels with setting the same thing in two different places. I suggest keeping the proposed PR changes until we deem a re-design is needed and then doing a database migration to get the values correct for downstream.

Reply:
I would just make the change as CIRRUS is relying on something that is non-changeable, so make it non-changeable in the grand challenge codebase. It is not duplication as the data representations are different. You can even skip the labels as these things are not used in forms.